Training6/training-6-sagas-login/src/components/ProtectedPage.js

```diff
@@ -0,0 +1,57 @@
+import React, { useEffect, useState } from "react";
+import { connect } from "react-redux";
+import { Redirect } from "react-router-dom";
+import { GetUser, LogOut } from "../actions/index";
+
+ProtectedPage.propTypes = {};
+
+function ProtectedPage({ GetUser, login, user, LogOut }) {
+  console.log("ProtectedPage -> user", user);
+  const tokenCurrent = login.token;
+  const [isBusy, setBusy] = useState(true);
+
+  useEffect(() => {
+    GetUser(tokenCurrent);
+    setTimeout(() => setBusy(false), 1000);
```

Comment:
why we need this setTimeout please explain me

Reply:
I had setTimeout to waiting for call state from the store. If not setTimeout, component render will noti undefined. To fix it, I added one payload "loaded" data in reducer GetUserReducer.